### PR TITLE
feat(ui/templates/variable.labels): export/import variable labels

### LINKED ISSUE #12935: Add labels to variable imports and exports
If a variable has a label, the exported variable template should have labels included in it. 
If an exported variable template has labels included- the labels should be created if they do not already exist, and added to the variable. 

- [ ] ensure client updates for variable labels: https://github.com/influxdata/influxdb2-js/pull/82
- [ ] Update to `influxdb-js` to `0.2.57` or higher for variable preferably `0.2.58`. The label relationship exporting currently expects `ILabel` so changes in the upcoming `0.2.58` will be needed. 
- [ ] Serialize `variable.labels[]` as part of  `relationships` and `includes` similar to `dashboard.labels[]` for exporting
- [ ] Add imported variable labels to variable.

### PR DESCRIPTION
**Pull request by OfTheDelmer**
Closes #12935 

_Briefly describe your proposed changes:_

-  Add tests for dashboards exporting
   - update dashboard exporting to include variable labels
- Update variable exporting to include labels
- Update importing to create labels for associated to variables.
    - Add associated labels to imported labels
    - Add labels to variables in imported dashboards

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [ ] Rebased/mergeable
  - [ ] Tests pass
  - [ ] http/swagger.yml updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/templates/utils/index.ts [hunk 1]
```diff
@@ -58,3 +55,20 @@ export const findVariablesToCreate = (
...
+}) => !!resource.relationships && !!resource.relationships[TemplateType.Label]
+
+export const getLabelRelationships = (resource: {
+  relationships?: Relationships
+}): LabelRelationship[] => {
+  if (!hasLabelsRelationships(resource)) {
+    return []
+  }
+
+  return [].concat(resource.relationships[TemplateType.Label].data)
```
Comment (OfTheDelmer) [inline ui/src/templates/utils/index.ts]: this was just a way to ensure if it's one or more labels it always an array of relationships

File: ui/src/shared/utils/resourceToTemplate.ts [hunk 2]
```diff
@@ -210,18 +229,25 @@ export const dashboardToTemplate = (
 
   const dashboardAttributes = _.pick(dashboard, ['name', 'description'])
 
-  const labels = getDeep<Label[]>(dashboard, 'labels', [])
-  const includedLabels = labels.map(l => labelToIncluded(l))
-  const relationshipsLabels = labels.map(l => labelToRelationship(l))
+  const dashboardLables = getDeep<Label[]>(dashboard, 'labels', [])
```
Comment (ischolten) [inline ui/src/shared/utils/resourceToTemplate.ts]: dashboardLabels not dashboardLables

File: ui/src/templates/api/index.ts [hunk 3]
```diff
@@ -46,49 +47,61 @@ export const createDashboardFromTemplate = async (
...
   } = template
 
-  if (!data.relationships || !data.relationships[TemplateType.Label]) {
-    return
+  const labeledResources = [data, ...included].filter(r =>
+    hasLabelsRelationships(r)
+  )
+
+  if (_.isEmpty(labeledResources)) {
+    return {}
```
Comment (ischolten) [inline ui/src/templates/api/index.ts]: could this be just a return

File: ui/src/templates/api/index.ts [hunk 4]
```diff
@@ -97,14 +110,17 @@ const createLabelsFromTemplate = async <T extends TemplateBase>(
   )
 
   const createdLabels = await client.labels.createAll(labelsToCreate)
+  const allLabels = [...createdLabels, ...existingLabels]
+
+  const labelMap: LabelMap = {}
 
-  // IDs of newly created labels that should be added to dashboard
-  const createdLabelIDs = createdLabels.map(l => l.id || '')
+  includedLabels.forEach(label => {
```
Comment (ischolten) [inline ui/src/templates/api/index.ts]: is there a reason for using a foreach rather than a reduce?

Comment (OfTheDelmer) [reply ui/src/templates/api/index.ts]: Juggling the accumulator and having to return the `labelMap` on each iteration when it just key-val update seemed like more overhead.

File: ui/src/templates/api/index.ts (hunk 3 above)
Comment (OfTheDelmer) [reply ui/src/templates/api/index.ts]: I wanted to maintain the return type of `LabelMap`. If it's empty the other helpers won't utilize it but it avoids having to add guards for `undefined`.

**Pull request merged**